core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1539,4 +1538,86 @@ public TaskDef getTaskDefinition(Task task) {
                             return new TerminateWorkflowException(reason);
                         }));
     }
+
+    /**
+     * Update parent Workflow status based on Subworkflow state.
+     * @param subWorkflow
+     */
+    public void updateParentWorkflow(Workflow subWorkflow) {
+        Workflow parent = executionDAOFacade.getWorkflowById(subWorkflow.getParentWorkflowId(), false);
+        WorkflowDef parentDef = Optional.ofNullable(parent.getWorkflowDefinition())
+                .orElseGet(() -> metadataDAO.getWorkflowDef(parent.getWorkflowName(), parent.getWorkflowVersion())
+                        .orElseThrow(() -> new ApplicationException(NOT_FOUND, String.format("Unable to find parent workflow definition for %s", parent.getWorkflowId())))
+                );
+        LOGGER.debug("Evaluating parent workflow: {} for sub-workflow: {}", subWorkflow.getParentWorkflowId(), subWorkflow.getWorkflowId());
+
+        Task parentWorkflowTask = executionDAOFacade.getTaskById(subWorkflow.getParentWorkflowTaskId());
+        // On workflow complete or terminate..
+        if (subWorkflow.getStatus().isTerminal()) {
+            // If parent is FAILED and the sub workflow task in parent is FAILED, we want to resume them
+            if (subWorkflow.getStatus().isSuccessful() && StringUtils.isBlank(parentDef.getFailureWorkflow()) && parent.getStatus() == WorkflowStatus.FAILED && parentWorkflowTask.getStatus() == FAILED) {
+                LOGGER.debug("Subworkflow: {} is successful, resetting failed parent workflow: {}, and Subworkflow task: {} status to IN_PROGRESS",
+                        subWorkflow.getWorkflowId(), parent.getWorkflowId(), subWorkflow.getParentWorkflowTaskId());
+                parentWorkflowTask.setStatus(IN_PROGRESS);
+                executionDAOFacade.updateTask(parentWorkflowTask);
+                parent.setStatus(WorkflowStatus.RUNNING);
+                executionDAOFacade.updateWorkflow(parent);
+                decide(parent.getWorkflowId());
+            } else if (parentWorkflowTask.getStatus() == IN_PROGRESS) {
+                LOGGER.debug("Subworkflow: {} is {}, updating parent workflow: {}",
+                        subWorkflow.getWorkflowId(), subWorkflow.getStatus().name(), parent.getWorkflowId());
+                SubWorkflow subWorkflowTask = new SubWorkflow();
+                subWorkflowTask.execute(subWorkflow, parentWorkflowTask, this);
+                // Keep Subworkflow task's data consistent with Subworkflow's.
+                if (parentWorkflowTask.getExternalOutputPayloadStoragePath() != null && !parentWorkflowTask.getOutputData().isEmpty()) {
```

Comment:
What is the case where the parent workflow task is expected to have output data in external storage while remaining in progress? Shouldn't the output data only get populated when the sub workflow task completes?